src/nlsolve/nlsolve.jl

```diff
@@ -61,7 +61,7 @@ function nlsolve!(nlsolver::AbstractNLSolver, integrator::DiffEqBase.DEIntegrato
 
         # check divergence (not in initial step)
         if iter > 1
-            θ = ndz / ndzprev
+            θ = nlsolver.prev_θ = max(0.3 * nlsolver.prev_θ, ndz / ndzprev)
```

Comment:
```suggestion
        # check divergence (using previous iterations divergance at the initial step)
        prev_θ = hasproperty(nlsolver, :prev_θ) ? nlsolver.prev_θ :  one(ndz)
        if iter > 1
            θ = prev_θ = max(0.3 * prev_θ, ndz / ndzprev)
```
This (combined with the fix on line 88) will fix the CI issue. The problem is that delay diffeq for better or worse (worse) uses this same code with it's own NLSolver struct. Presumably, that struct should get the same optimization, but I'm not fully sure the math works out in the delay case, and this will preserve the status quo for them.

Reply:
But if you do this then `nlsolver.prev_θ` never updates? I think the right thing to do is just propagate this eveywhere.

Reply:
Is there a good reason why DellayDiffEq needs it's own solver at all? It does seem like the interface here might not be especially well defined...

Reply:
Maybe they could be unified a bit more. But DelayDiffEq is something quite special/different with its NLSolver, it uses it for fixed point iterations of the actual integration step which involve running an ODE integrator: https://github.com/SciML/DelayDiffEq.jl/blob/43201e7ac1169ab0c08bc3a41879e4445ce2c827/src/fpsolve/functional.jl#L96-L114